crates/ra_parser/src/grammar/params.rs

```diff
@@ -41,9 +44,16 @@ fn list_(p: &mut Parser, flavor: Flavor) {
     let m = p.start();
     p.bump();
     if flavor.type_required() {
+        attributes::outer_attributes(p);
         opt_self_param(p);
     }
     while !p.at(EOF) && !p.at(ket) && !(flavor.type_required() && p.at(T![...])) {
+        attributes::outer_attributes(p);
+
+        if p.at(T![...]) {
```

Comment:
this check is duplicated with the loop header. Let's remove `flavor.type_required() && p.at(T![...])` from the while condition and just breaks out of the loop? Note the `flavor` bit which I think is missing from this `if`